tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__Handler.java

```diff
@@ -61,23 +61,41 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             // TODO: handle command
 
             // Note: if communication with thing fails for some reason,
-            // indicate that by setting the status with detail information
+            // indicate that by setting the status with detail information:
             // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
             // "Could not control device at IP address x.x.x.x");
         }
     }
 
     @Override
     public void initialize() {
+        // logger.debug("Start initializing!");
         config = getConfigAs(${bindingIdCamelCase}Configuration.class);
 
-        // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
-        // Long running initialization should be done asynchronously in background.
+        // TODO: Initialize the thing.
+        // The framework requires you to leave this method quickly. Also, before leaving this method a thing status
+        // from one of ONLINE, OFFLINE or UNKNOWN must be set. This might already be the real thing status in case you
+        // can decide it directly.
+        // In case you can not decide the thing status directly (e.g. for long running initialization using WAN access
+        // or similar) you should set status OFFLINE here and then decide the real status asynchronously in the
+        // background.
 
-        // logger.debug("Start initializing!");
-        updateStatus(ThingStatus.ONLINE);
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            boolean initialized = true; // <background task with long running initialization here>
```

Comment:
`initialized` is misleading here - it should rather be `reachable` or `online` here as this is what you are trying to figure out.